lib/neo4j/core/cypher_session/adaptors/http.rb

```diff
@@ -16,7 +16,7 @@ def initialize(url, options = {})
           end
 
           def connect
-            @requestor = Requestor.new(@url, USER_AGENT_STRING, self.class.method(:instrument_request))
+            @requestor = Requestor.new(@url, USER_AGENT_STRING, self.class.method(:instrument_request), @options[:faraday_opts] = {})
```

Comment:
Nitpick: `faraday_options` would be better, I think

Also, did you mean to do `@options[:faraday_opts] ||= {}` here?  Otherwise it's going to always set / pass through an empty `Hash`.  You could also do `@options.fetch(:faraday_options, {})`